extensions/src/AWSSDK.Extensions.Configuration.AWSSystemsManager/AWSSystemsManagerConfigurationProvider.cs

```diff
@@ -0,0 +1,120 @@
+        /// </summary>
+        /// <param name="source">The <see cref="IConfigurationSource"/> used to retrieve values from AWS Systems Manager Parameter Store</param>
+        public AWSSystemsManagerConfigurationProvider(AWSSystemsManagerConfigurationSource source) : this(source, new AWSSystemsManagerProcessor())
+        {
+        }
+
+        public AWSSystemsManagerConfigurationProvider(AWSSystemsManagerConfigurationSource source, IAWSSystemsManagerProcessor awsSystemsManagerProcessor)
+        {
+            Source = source ?? throw new ArgumentNullException(nameof(source));
+            _awsSystemsManagerProcessor = awsSystemsManagerProcessor;
+
+            if (source.AwsOptions == null) throw new ArgumentNullException(nameof(source.AwsOptions));
+            if (source.Path == null) throw new ArgumentNullException(nameof(source.Path));
+
+            if (source.ReloadAfter != null)
+            {
+                ChangeToken.OnChange(() =>
+                {
+                    var cancellationTokenSource = new CancellationTokenSource(source.ReloadAfter.Value);
+                    var cancellationChangeToken = new CancellationChangeToken(cancellationTokenSource.Token);
+                    return cancellationChangeToken;
+                }, async () => await LoadAsync(true));
+            }
+        }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Loads the AWS Systems Manager Parameters.
+        /// </summary>
+        public override void Load() => LoadAsync(false).ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync(bool reload)
+        {
+            try
+            {
+                var path = Source.Path;
+                var awsOptions = Source.AwsOptions;
+                var parameters = await _awsSystemsManagerProcessor.GetParametersByPathAsync(awsOptions, path);
+
+                Data = ProcessParameters(parameters, path);
+
+                OnReload();
+            }
+            catch (Exception ex)
+            {
+                if (Source.Optional || reload) return;
+
+                var ignoreException = false;
+                if (Source.OnLoadException != null)
+                {
+                    var exceptionContext = new AWSSystemsManagerExceptionContext
+                    {
+                        Provider = this,
+                        Exception = ex
+                    };
+                    Source.OnLoadException(exceptionContext);
+                    ignoreException = exceptionContext.Ignore;
+                }
+
+                if (!ignoreException)
+                    throw;
+            }
+        }
+
+        private static string NormalizeKey(string key)
+        {
+            return key.Replace("/", ConfigurationPath.KeyDelimiter);
+        }
+
+        public static IDictionary<string, string> ProcessParameters(List<Parameter> parameters, string path) =>
+            parameters
+                .Select(parameter => new
```

Comment:
Please consider allowing something like a ```IParameterProcessor``` to be passed into the Provider so that we can customize this. You can make this the default implementation that can be created in an overloaded extension method.

Reply:
I had previously thought about doing this. I didn't do it in the initial implementation since the Param Store guidelines say to use "/" to organize the parameters, and I didn't want to make the initial version over complicated. However, I realize that users will create their own organization structures. I'll look at making this update over the weekend. I'll probably only add it to the source extension and not make a special extension for this. The inline extensions are for the more straight forward/default uses. Once it gets into the more complicated structures that what the source version is for.

example:
```csharp
.AddSystemsManager(source =>
{
    source.ParameterProcessor = MyCustomProcessor()
}
```